### ISSUE TITLE
The allow_metadata_inventory_take callback is handled multiple times when shiftclicking

### ISSUE THREAD

**Issue by dopik** [minetest/minetest#7239]
##### Issue type
 - Bug report 

##### Minetest version
```
0.4.16
```

##### Summary
When shiftclicking an 'overstacked' (more items then maxStackSize) ItemStack out of an node's inventory, the node's **allow_metadata_inventory_take** function is called multiple times and only one set of the items are actually gotten by the player.

##### Steps to reproduce

 - Place this examplenode in your world:
```LUA
minetest.register_node("testmod:testnode",{
	groups = {
		oddly_breakable_by_hand = 1
	},
	after_place_node = function(pos, placer, itemstack, pointed_thing)
		local meta = minetest.get_meta(pos)
		local inv = meta:get_inventory()
		
		inv:set_size("testinventory",8)
		meta:set_string("formspec", "size[8,5]list[context;testinventory;0,0;8,1]list[current_player;main;0,1.25;8,4]listring[]")
	end,
	allow_metadata_inventory_take = function(pos, listname, index, stack, player)
		minetest.chat_send_all("AllowOutWasCalled")
		return 1
	end,
})

``` 
 - Give yourself some items through the /give command (take more then the item usually stacks to; e.g.  for stone at least 100)
 - Put the items into the node's inventory
 - Shiftclick the items out and count the number of chatmessages and the resulting number of total items (so if you had 100 items in and got one out, 99 should remain in the node and 1 should be in your inventory. 99 + 1 = 100)
 - Take the itmes out without using shift for comparison

**Comment by sfan5** [comment]
Nowhere does the documentation guarantee that allow_metadata_inventory_* is only called once.
Not a bug I'd say.

**Comment by dopik** [comment]
Yes, you're right, the documentation doesn't say that it is called only once. But it seems weird that only the *take callback does this **and** only when shiftclicking **and** then also only when overstacked... Also it seems buggy or at least not intended that it then deletes items as I discribed
